### ISSUE TITLE
[Feature Request] WebSocket connect timeout

### ISSUE THREAD

**Issue by bobaoapae** [airsdk/Adobe-Runtime-Support#3350]
WebSocket connect method should accept a timeout parameter or some get/set to better implement connection/reconnection logics.

Something like.

```
var websocket:WebSocket = new WebSocket();
websocket.connect('address',null,3000);

```
or

```

var websocket:WebSocket = new WebSocket();
websocket.timeout = 3000;
websocket.connect('address',null);

```
timeout should dispatch ioError if websocket connection could not be established in the time specified.

**Comment by ajwfrost** [comment]
Good point - there's already something like this in the `Socket` object..
https://airsdk.dev/reference/actionscript/3.0/flash/net/Socket.html#timeout
